### ISSUE TITLE
exq worker exception not sent

### ISSUE THREAD

**Issue by StephaneRob** [getsentry/sentry-elixir#337]
### Environment

Tested with two configurations

* Elixir version (elixir -v): 1.7.3 / 1.8.1
* Erlang/OTP version (erl): 21.3.3
* Sentry version (mix deps): 6.4.2 / 7.0.6

### Description

Exceptions raised on [exq](https://github.com/akira/exq) workers are not sent to sentry when running app via a release distillery.

By running the app with mix, exceptions are correctly sent.

**Comment by mitchellhenke** [comment]
Hi @StephaneRob, thanks for opening an issue!

For the exceptions being raised, are they exceptions that would crash the process, or are you capturing them in the job and sending them manually?

exq offers a middleware hook when a job fails in `after_failed_work`.  Is that something you're using?

I'm unfortunately not super familiar with distillery deployments, and am unsure what would cause such an issue.  Do other Sentry exceptions work, and it's just exq workers with the problem?

**Comment by StephaneRob** [comment]
Hi @mitchellhenke 

Actually I finally catch manually exceptions in `after_failed_work` hook. But I didn't need anything when running the app with `mix phx.server`.

It looks like the release assembled by distillery doesn't embed something useful to catch automatically the exception. (it does not depends on the env, tested with dev or prod release)

Yes it's only on exq worker.

**Comment by mitchellhenke** [comment]
It seems this issue is outside of Exq, Distillery, and Sentry.  I've opened up a PR in Elixir here that should fix the issue: https://github.com/elixir-lang/elixir/pull/9020

Should it be merged, a new Elixir version will need to be released, etc. before the fix is available

In the meantime, it's probably best to rely on the `after_failed_work` hook and send events manually if you are running under a Distillery release:

```elixir
defmodule ErrorReports.ExqSentryMiddleware do
  @behaviour Exq.Middleware.Behaviour

  def before_work(pipeline) do
    pipeline
  end

  def after_processed_work(pipeline) do
    pipeline
  end

  def after_failed_work(pipeline) do
    {error, stacktrace} = pipeline.assigns.error
    Sentry.capture_exception(error, [stacktrace: stacktrace])
    pipeline
  end
end
```

**Comment by StephaneRob** [comment]
Thanks for the search and for the info.
We will keep the manual `capture_exception` into `after_failed_work` until the fix on the elixir side.

**Comment by mitchellhenke** [comment]
Sounds good, thank you again for reporting this issue 🙂 

I'm going to close this issue, but feel free to comment back if something comes up and it needs to be reopened.

**Comment by StephaneRob** [comment]
Thank you for your help @mitchellhenke !

**Comment by mitchellhenke** [comment]
A bit behind on sharing it, but the fix was released in [Elixir 1.8.2](https://github.com/elixir-lang/elixir/blob/v1.8/CHANGELOG.md#v182-2019-05-11)
